### ISSUE TITLE
More helpful error message when proto_toolchains not registered

### ISSUE THREAD

**Issue by nathantchan** [bazelbuild/rules_go#996]
If I forget to include proto_register_toolchains() in my WORKSPACE file, bazel will spit out the following stacktrace when I try compiling a go_proto_library:
```
java.lang.IllegalStateException: CONFIGURED_TARGET://third_party/go/src/vendor/google.golang.org/grpc:go_default_library 54cd02274c0055dec82686a17b4c1d1e (1764287266 1042919375) -> <ErrorInfo exception=com.google.devtools.build.lib.skyframe.ToolchainResolutionFunction$NoToolchainFoundException: no matching toolchain found for @io_bazel_rules_go//proto:go_proto rootCauses={ToolchainResolutionKey{configuration=54cd02274c0055dec82686a17b4c1d1e, toolchainType=@io_bazel_rules_go//proto:go_proto, targetPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64")), struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux"))], label = Label("@bazel_tools//platforms:target_platform")), execPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64")), struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux"))], label = Label("@bazel_tools//platforms:host_platform"))}} cycles=[]>
	at com.google.common.base.Preconditions.checkState(Preconditions.java:721)
	at com.google.devtools.build.lib.util.Preconditions.checkState(Preconditions.java:238)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:412)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:278)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:600)
	at com.google.devtools.build.lib.buildtool.BuildTool.runAnalysisPhase(BuildTool.java:505)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:214)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:341)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:67)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:600)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:351)
	at com.google.devtools.build.lib.runtime.CommandExecutor.exec(CommandExecutor.java:58)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:850)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$2100(GrpcServerImpl.java:109)
	at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:915)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
java.lang.IllegalStateException: CONFIGURED_TARGET://third_party/go/src/vendor/google.golang.org/grpc:go_default_library 54cd02274c0055dec82686a17b4c1d1e (1764287266 1042919375) -> <ErrorInfo exception=com.google.devtools.build.lib.skyframe.ToolchainResolutionFunction$NoToolchainFoundException: no matching toolchain found for @io_bazel_rules_go//proto:go_proto rootCauses={ToolchainResolutionKey{configuration=54cd02274c0055dec82686a17b4c1d1e, toolchainType=@io_bazel_rules_go//proto:go_proto, targetPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64")), struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux"))], label = Label("@bazel_tools//platforms:target_platform")), execPlatform=struct(constraints = [struct(constraint = struct(label = Label("@bazel_tools//platforms:cpu")), label = Label("@bazel_tools//platforms:x86_64")), struct(constraint = struct(label = Label("@bazel_tools//platforms:os")), label = Label("@bazel_tools//platforms:linux"))], label = Label("@bazel_tools//platforms:host_platform"))}} cycles=[]>
	at com.google.common.base.Preconditions.checkState(Preconditions.java:721)
	at com.google.devtools.build.lib.util.Preconditions.checkState(Preconditions.java:238)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:412)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:278)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:600)
	at com.google.devtools.build.lib.buildtool.BuildTool.runAnalysisPhase(BuildTool.java:505)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:214)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:341)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:67)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:600)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:351)
	at com.google.devtools.build.lib.runtime.CommandExecutor.exec(CommandExecutor.java:58)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:850)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$2100(GrpcServerImpl.java:109)
	at com.google.devtools.build.lib.server.GrpcServerImpl$2.lambda$run$0(GrpcServerImpl.java:915)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```
I think it would be more helpful if we could capture this stacktrace and print a more helpful error message like:
"Building a go_proto_library rule requires registering proto toolchains. Please add:
load("@io_bazel_rules_go//proto:def.bzl", "proto_register_toolchains")
proto_register_toolchains()
to your WORKSPACE"

**Comment by jayconrod** [comment]
This looks like #941, which is caused by bazelbuild/bazel#3928. That was fixed, and should be working in Bazel 0.8, which should be released in a week or two. There's already an RC out, if you want to try that.

I'll close this as a duplicate. Unfortunately, there's nothing we can do in Skylark to catch Bazel crashes like this. #942 suggests having Gazelle warn when `proto_register_toolchains` is not called, but I haven't had time to work on that yet.

**Comment by jayconrod** [comment]
This looks like #941, which is caused by bazelbuild/bazel#3928. That was fixed, and should be working in Bazel 0.8, which should be released in a week or two. There's already an RC out, if you want to try that.

I'll close this as a duplicate. Unfortunately, there's nothing we can do in Skylark to catch Bazel crashes like this. #942 suggests having Gazelle warn when `proto_register_toolchains` is not called, but I haven't had time to work on that yet.
